### ISSUE TITLE
Error: node_modules/@firebase/auth/dist/auth-public.d.ts:3225:23 - error TS2304: Cannot find name 'FactorId_2'.

### ISSUE THREAD

**Issue by jnizet** [firebase/firebase-js-sdk#7174]
### [REQUIRED] Describe your environment

  * Operating System version: macOS 13.1
  * Browser version: NA
  * Firebase SDK version: 9.19.0
  * Firebase Product: auth

#### Steps to reproduce:
- Try to compile an Angular project using Firebase 9.19.0
- The compilation fails with the error

```
Error: node_modules/@firebase/auth/dist/auth-public.d.ts:3225:23 - error TS2304: Cannot find name 'FactorId_2'.
```

The auth-public.d.ts file does indeed seem to reference a `FactorId_2` symbol that doesn't exist.

It's documented as excluded at line 1160 of the file:

```
/* Excluded from this release type: FactorId_2 */
```

**Comment by eun-choi** [comment]
+1

**Comment by dwyfrequency** [comment]
Can you provide a minimal reproducible repo?

**Comment by duncanchiu409** [comment]
+1 
`Error: node_modules/@firebase/auth/dist/auth-public.d.ts:3225:23 - error TS2304: Cannot find name 'FactorId_2'.

3225     static FACTOR_ID: FactorId_2;`

**Comment by duncanchiu409** [comment]
I am using firebase version: ^9.6.7 . Had the same issue

**Comment by duncanchiu409** [comment]
This is my reproduced error repo. Hope helps. Pretty simple to reproduce the problem.
https://github.com/duncanchiu409/firebase-issue7174

**Comment by jnizet** [comment]
Here's mine (this PR specifically, which upgrades to the latest version: https://github.com/jnizet/maison-solidaire/pull/147).

Steps to reproduce:
- clone the repo and checkout that branch
- yarn
- yarn build

**Comment by PrajinThunoli** [comment]
Same for me too
node_modules/@firebase/auth/dist/auth-public.d.ts:3225:23 - error TS2304: Cannot find name 'FactorId_2'.

3225     static FACTOR_ID: FactorId_2;

Did a npm update and build this issue came up

**Comment by hassan-badat** [comment]
A temporary work around is to add the value:

`"skipLibCheck": true`

to your tsconfig.json

**Comment by malik-devop** [comment]
i can't build my docker image because i have the same error:

Error: node_modules/@firebase/auth/dist/auth-public.d.ts:3225:23 - error TS2304: Cannot find name 'FactorId_2'.
 
how can i solve this problem ?

**Comment by Monuk0** [comment]
@dwyfrequency I am facing similar issue in my project can you please explain me what exactly is minimal minimal reproducible repo, my project in angular 13 could you please help me. to solve this issue. I really great thx. full to you.

**Comment by DwieDima** [comment]
same here.
For now i'm downgrading to `"firebase": "9.9.1"` until issue is resolved

**Comment by JakobMadsen717** [comment]
I think downgrading to firebase@9.18.0 would work fine

**Comment by DimaIgorevich** [comment]
But this issue also repro on @angular/fire lib because has dependency.

**Comment by pietroow** [comment]
> A temporary work around is to add the value:
> 
> `"skipLibCheck": true`
> 
> to your tsconfig.json

Doing that does not avoid the error.
You still have the issue at runtime

> I think downgrading to firebase@9.18.0 would work fine

Not working, this version still having the same implementation.
`static FACTOR_ID: FactorId_2;`

**Comment by henriquesschmitt** [comment]
Same error here, using "firebase": "^9.19.0"

**Comment by dpsifr** [comment]
Seeing the same problem and it has left me dead in the water and not able to even start or build our project.

**Comment by hsubox76** [comment]
We'll try to get a fix out today but you can just roll back to the previous version of Firebase (9.18.0) in the meantime.

**Comment by hsubox76** [comment]
Published 9.19.1, which should have the fix.

For future users who come by this issue: Do not use 9.19.0, use 9.19.1 or higher.

**Comment by RichardGuesso** [comment]
@hsubox76 its work!!!
